demos/stardist_demo/python/stardist_demo.py

```diff
@@ -0,0 +1,373 @@
+                            default='0', type=str)
+    infer_args.add_argument('-nthreads', '--num_threads', default=None, type=int,
+                            help='Optional. Number of threads to use for inference on CPU (including HETERO cases).')
+    args = parser.parse_args()
+
+    if args.input.endswith('.ndpi'):
+        img = OpenSlideImageReader(args.input)
+    else:
+        img = BFImageReader(args.input)
+
+    log.info('Initializing Inference Engine...')
+    ie = IECore()
+
+    plugin_config = get_plugin_configs(args.device, args.num_streams, args.num_threads)
+
+    log.info('Loading network...')
+
+    model = StarDistModel(ie, args)
+
+    detector_pipeline = AsyncPipeline(ie, model, plugin_config,
+                                      device=args.device, max_num_requests=args.num_infer_requests)
+
+    tile_id = (0, 0)
+    ready_tile_id = (0, 0)
+    num_tiles_x = ceil(img.width / args.tile_size)
+    num_tiles_y = ceil(img.height / args.tile_size)
+    log.info('Number of tiles: {}'.format(num_tiles_x * num_tiles_y))
+
+    if not args.no_show:
+        img_preview, preview_scale = img.get_preview(max_width=256)
+
+    def next_tile(tile_id):
+        x, y = tile_id
+        if x < num_tiles_x - 1:
+            return (x + 1, y)
+        elif y < num_tiles_y - 1:
+            return (0, y + 1)
+        else:
+            return None
+
+    start = time.time()
+
+    while True:
+        if detector_pipeline.callback_exceptions:
+            raise detector_pipeline.callback_exceptions[0]
+        # Process all completed requests
+        results = detector_pipeline.get_result(ready_tile_id)
+        if results:
+            contours, frame_meta = results
+            tile = frame_meta['tile']
+            if not args.no_show:
+                if tile.shape[2] == 1:
+                    tile = np.repeat(tile, 3, axis=-1)
+                else:
+                    tile = cv.cvtColor(tile, cv.COLOR_RGB2BGR)
+
+                # Draw detections
+                tile = cv.drawContours(tile, contours, -1, (0, 0, 255))
+
+                # Draw a preview
+                if img_preview is not None:
+                    offset = tile.shape[1] - img_preview.shape[1]
+                    tile[0:img_preview.shape[0], offset:] = img_preview
+                    preview_roi_sz = int(args.tile_size * preview_scale)
+                    preview_roi_x = int(ready_tile_id[0] * preview_roi_sz)
+                    preview_roi_y = int(ready_tile_id[1] * preview_roi_sz)
+                    cv.rectangle(tile, (offset + preview_roi_x, preview_roi_y, preview_roi_sz, preview_roi_sz),
+                                 (0, 0, 255))
+
+                cv.imshow('StarDist with OpenVINO', tile)
+                key = cv.waitKey(1)
+
+                ESC_KEY = 27
+                # Quit.
+                if key in {ord('q'), ord('Q'), ESC_KEY}:
+                    break
```

Comment:
Since the demo is just like others now, can you add `Presenter` https://github.com/openvinotoolkit/open_model_zoo/blob/7577edeaa83d8bc2d30f200cdc786fdfc1d4ddb1/demos/object_detection_demo/python/object_detection_demo.py#L293
drawing CPU and memory usage and handle key press for it here. And a demo arg to enable them at startup.

You can create an instance of `Presenter` out of main loop because the size of frame is known in advance.